warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -1657,7 +57,7 @@ public static boolean isIndexed(JexlNode node, ShardQueryConfiguration config) {
         Preconditions.checkNotNull(indexedFieldsDatatypes);
```

Comment:
Can you briefly walk me through the thought process behind having two JexlASTHelpers? If I'm going to add a method to a JexlASTHelper, what's the criteria for placing a method in the core-query instance vs. the query-core instance?

Reply:
If you look at datawave.query.jexl.JexlASTHelper you will notice that all of the methods in this class (aside from the two at the end added by Charlie after I made this PR) depend on at least one class in datawave-query-core.  Some examples are RangeStream, JexlArgumentDescriptorFactory, and IndexStatsClient).  Those classes are unrelated to jexl, and thus don't really belong in the datawave-core-jexl module.  I also don't want datawave-core-jexl to have a dependency on datawave-query-core, so the only option would be to move those classes into yet another separate module (and deal with all of the implications of that) or extend JexlASTHelper in datawave-query-core to 1) maintain backwards compatibility, and 2) provide a JexlASTHelper class which has access to those constructs.  

I think an argument could be made that JexlASTHelper has become somewhat of a dumping ground, and that those methods in datawave.query.jexl.JexlASTHelper could and probably should be moved into a class other than JexlASTHelper.  If we ever achieve that, then there will be back to having a single JexlASTHelper.